### DISCUSSION TITLE
Autocompletion is short

### DISCUSSION THREAD

**Discussion by dbian** [rjmacarthy/twinny#183] [Q&A] [answered]
Compared to tabby and copilot, twinny can not generate long codes, for example 5 lines of codes.

Good job, by the way. I appreciate your hard work.

**Comment by DynDux** [comment]
Did you go through the settings of twinny?
Maybe "Max Lines" or "Num Predict Fim" need adjustments.

**Comment by rjmacarthy** [comment, marked as answer]
Yes, please try to tweak the settings. Enable multiline completions, num predict FIM, and max lines.

**Comment by dbian** [comment]
![image](https://github.com/rjmacarthy/twinny/assets/1283828/d735b3a7-504d-4977-b453-6c9f3f7759e2)

Thank you all. I tried, but still produce only 1-2 lines of code.

My settings:

![image](https://github.com/rjmacarthy/twinny/assets/1283828/404ca4ae-c468-472f-8bde-be72e8f56d9b)

**Comment by rjmacarthy** [comment]
Sometimes the model will send a stop token or the provider will end the completion stream, can we confirm it's not the case? Please update twinny and the provider to the latest versions and report back again. Thanks.
